randovania/games/fusion/exporter/patch_data_factory.py

```diff
@@ -216,6 +217,43 @@ def _create_nav_text(self) -> dict:
             }
         return nav_text_json
 
+    def _create_credits_text(self) -> dict:
+        credits_array = []
+        spoiler_dict = []
+        for player_index, patches in self.description.all_patches.items():
+            for pickup_index, target in patches.pickup_assignment.items():
+                if target.player != self.players_config.player_index:
+                    continue
+                region_list = default_database.game_description_for(patches.configuration.game).region_list
+                pickup_category = target.pickup.pickup_category
+                if pickup_category.hinted_as_major or pickup_category.is_key:
+                    player_name = None
+                    if self.players_config.is_multiworld:
+                        player_name = self.players_config.player_names[player_index]
+                    spoiler_dict.append(
+                        {
+                            "Item": target.pickup.name,
+                            "Location": {
+                                "World": player_name,
+                                "Region": region_list.region_name_from_node(
+                                    region_list.node_from_pickup_index(pickup_index), True
+                                ),
+                                "Area": region_list.nodes_to_area(
+                                    region_list.node_from_pickup_index(pickup_index)
+                                ).name,
+                            },
+                        }
+                    )
+        for item in spoiler_dict:
```

Comment:
you should group the spoiler_dict by item, so that you can have
```
MORPH BALL
Sector 1 crab rave
Sector 2 Kago Room
Sector 4 C-Cache
```

Reply:
I should... how?

Reply:
By using the spoiler_dict as an actual dictionary and not as a list.

Reply:
can I get a hint